clippy_lints/src/strings.rs

```diff
@@ -154,7 +154,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StringLitAsBytes {
                     if let LitKind::Str(ref lit_content, style) = lit.node {
                         let callsite = snippet(cx, args[0].span.source_callsite(), r#""foo""#);
                         let expanded = if let StrStyle::Raw(n) = style {
-                            let term = (0..n).map(|_| '#').collect::<String>();
+                            let term = "#".repeat(n as usize);
```

Comment:
```suggestion
                            let term = "#".repeat(usize::from(n));
```
I imagine it could trigger `clippy::cast_lossless`, this version results in the same assembly with opt-level >= 2: https://rust.godbolt.org/z/rE7G79

BTW according to LLVM MCA `repeat` is two times faster.